1.1.2

skupr-anaconda opened the pull request.

skupr-anaconda committed:
Release 1.1.2

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_e4gdguy9mp/clone/recipe/meta.yaml:1: cython_needs_compiler: Cython generates C code, which will need to be compiled
ERROR: /tmp/abs_e4gdguy9mp/clone/recipe/meta.yaml:52: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

skupr-anaconda committed:
Add run_test.py

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_95zl96nzdf/clone/recipe/meta.yaml:1: cython_needs_compiler: Cython generates C code, which will need to be compiled
ERROR: /tmp/abs_95zl96nzdf/clone/recipe/meta.yaml:56: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

skupr-anaconda committed:
Add a patch

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_4023308rhd/clone/recipe/meta.yaml:26: cython_needs_compiler: Cython generates C code, which will need to be compiled
ERROR: /tmp/abs_4023308rhd/clone/recipe/meta.yaml:26: cython_needs_compiler: Cython generates C code, which will need to be compiled
ERROR: /tmp/abs_4023308rhd/clone/recipe/meta.yaml:61: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

skupr-anaconda committed:
Fix dependencies

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_7ecl_5hc6n/clone/recipe/meta.yaml:22: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_7ecl_5hc6n/clone/recipe/meta.yaml:53: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

akabanovs committed:
Build v 1.1.3:
 - version and sha256 set up.
 - Seems like binary_has_prefix_files not needed anymore.
 - script_env with env variable added to skip downloading and building cmdstan. Instead, use local available.
 - missing_dso_whitelist added to suppress warnings. tbb and libcxx are implicitly added and are available.
 - cmdstan pinned to a specific version in host section.
 - importlib_resources are now required in run dependencies.
 - test section and run_test.py reviewed and updated to satisfy linter.

akabanovs committed:
Adding the following:
 - binary_has_prefix_files attribute
 - removed missing_dso_whitelist. Instead, added tbb and libcxx dependencies

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_45zxweyxv1/clone/recipe/meta.yaml:30: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_45zxweyxv1/clone/recipe/meta.yaml:31: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Adding build field with c++ compiler

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_33x_iny4_e/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Adding cmdstan as a dependency

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_cfdu59bl02/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_cfdu59bl02/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Skipping s390x due to ephem package unavailability

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_a9g1naz1_q/clone/recipe/meta.yaml:34: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_a9g1naz1_q/clone/recipe/meta.yaml:35: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

akabanovs committed:
Test for windows only

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_b9z8oizl3b/clone/recipe/meta.yaml:36: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_b9z8oizl3b/clone/recipe/meta.yaml:37: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
Errors were found

</details>

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -1,71 +1,62 @@
-    - tqdm >=4.36.1
+    - setuptools
+    - wheel
+    - tbb-devel
+    - cmdstan
```
```
tbb-devel 2021.8.0
cmdstan 2.31.0
```

akabanovs committed:
Another test for win64
